control-planes.tf

```diff
@@ -20,9 +20,19 @@ resource "outscale_keypair" "control-planes" {
   public_key = tls_private_key.control-planes[count.index].public_key_openssh
 }
 
+resource "outscale_public_ip" "control-planes" {
+  count = var.public_cloud ? var.control_plane_count : 0
+}
+
+resource "outscale_public_ip_link" "control-planes" {
+  count     = var.public_cloud ? var.control_plane_count : 0
+  vm_id     = outscale_vm.control-planes[count.index].vm_id
+  public_ip = outscale_public_ip.control-planes[count.index].public_ip
+}
+
 resource "outscale_security_group" "control-plane" {
   description = "Kubernetes control-planes (${var.cluster_name})"
-  net_id      = outscale_net.net.net_id
+  net_id      = var.public_cloud ? "" : outscale_net.net[0].net_id
```

Comment:
why sometimes var.public_cloud ? "" and other it is var.public_cloud ? nil for net_id